src/main/java/frc/robot/subsystems/DriveBaseSubsystem.java

```diff
@@ -9,42 +9,50 @@
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import edu.wpi.first.wpilibj.Joystick;
 import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.ADXRS450_Gyro;
 
 
 public class DriveBaseSubsystem extends SubsystemBase {
 
   private final Joystick m_driverJoystick;
-  private final MotorController[] m_motorControllers = new MotorController[6];
+  private final MotorController[] m_motorControllers;
   private final DifferentialDrive m_differentialDrive;
-
+  //public static ADIS16448_IMU m_gyro; Non-native gyro, might use later
+  public static ADXRS450_Gyro m_gyro;
+  private final DifferentialDriveOdometry m_odometry;
   
+
   public DriveBaseSubsystem(Joystick joystick) {  
     m_driverJoystick = joystick;
+    m_motorControllers = new MotorController[4];
+    m_odometry = new DifferentialDriveOdometry(m_gyro.getRotation2d());
+    m_gyro = new ADXRS450_Gyro();
+    m_differentialDrive = new DifferentialDrive(m_motorControllers[Constants.kDriveLeftFrontIndex].getSparkMax(), 
```

Comment:
This needs to be done after we've already created motor controllers for the Left and Right front wheels. As is, this will throw a null reference exception when run.